data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileScheduler.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.dynamodb.export;
+
+import io.micrometer.core.instrument.Counter;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.EnhancedSourceCoordinator;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.SourcePartition;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.partition.DataFilePartition;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.partition.GlobalState;
+import org.opensearch.dataprepper.plugins.source.dynamodb.model.LoadStatus;
+import org.opensearch.dataprepper.plugins.source.dynamodb.model.TableInfo;
+import org.opensearch.dataprepper.plugins.source.dynamodb.model.TableMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiConsumer;
+
+
+public class DataFileScheduler implements Runnable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DataFileScheduler.class);
+
+    private final AtomicInteger numOfWorkers = new AtomicInteger(0);
+
+    private static final int MAX_JOB_COUNT = 2;
+    private static final int DEFAULT_LEASE_INTERVAL_MILLIS = 30_000;
+
+    static final String EXPORT_FILE_SUCCESS_COUNT = "exportFileSuccess";
+
+
+    private final EnhancedSourceCoordinator coordinator;
+
+    private final ExecutorService executor;
+
+    private final DataFileLoaderFactory loaderFactory;
+
+    private final PluginMetrics pluginMetrics;
+
+
+    private final Counter exportFileSuccessCounter;
+
+
+    public DataFileScheduler(EnhancedSourceCoordinator coordinator, DataFileLoaderFactory loaderFactory, PluginMetrics pluginMetrics) {
+        this.coordinator = coordinator;
+        this.pluginMetrics = pluginMetrics;
+        this.loaderFactory = loaderFactory;
+
+
+        executor = Executors.newFixedThreadPool(MAX_JOB_COUNT);
+
+        this.exportFileSuccessCounter = pluginMetrics.counter(EXPORT_FILE_SUCCESS_COUNT);
+    }
+
+    private void processDataFilePartition(DataFilePartition dataFilePartition) {
+        String exportArn = dataFilePartition.getExportArn();
+        String tableArn = getTableArn(exportArn);
+
+        TableInfo tableInfo = getTableInfo(tableArn);
+
+        Runnable loader = loaderFactory.createDataFileLoader(dataFilePartition, tableInfo);
+        CompletableFuture runLoader = CompletableFuture.runAsync(loader, executor);
+        runLoader.whenComplete(completeDataLoader(dataFilePartition));
+        numOfWorkers.incrementAndGet();
+    }
+
+    @Override
+    public void run() {
```

Comment:
It seems we have this one thread which then creates two other threads. Can we eliminate this "scheduler" thread somehow? Perhaps just run two threads that each acquire a partition and then process it.